### ISSUE TITLE
What Visual Studio version to use?

### ISSUE THREAD

**Issue by taxfromdk** [elliotwoods/ofxBlackmagic2#5]
Hi Guys,

I  cant seem to build the project under Visual Studio Express 2012. 

The solution seems to be setup for VS2013, which is difficult for me as OpenFrameworks depend on VS2012.

What I did: 

I download th openframeworks for VS2012 package as zip file and place the ofxBlackmagic2 under the "other" folder within. Open the project file and build.

The error I get is:

Error   1   error MSB8020: The builds tools for v120 (Platform Toolset = 'v120') cannot be found. To build using the v120 build tools, either click the Project menu or right-click the solution, and then select "Update VC++ Projects...". Install v120 to build using the v120 build tools.  C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\V110\Microsoft.Cpp.Platform.targets   44  5   ofxBlackmagicLib

The v120 toolset points toward VS2013, and I cant change the toolset. 

I also get a popup saying unspecified error when I change the configuration, making  me suspect that the solution file is corrupt or something.

A pointer on where to go from here will be higly appreciated.

Kind regards

Jesper

**Comment by elliotwoods** [comment]
You just need to change the platform toolset. The solution isn't corrupt as far as I know (I'm using it regularly)

Make sure:
* change platform toolset to v110 on all projects in solution (openFrameworks should already be set)
* clean solution and rebuild

Please post screenshot of error (you can paste directly here). 

Which example are you trying?

**Comment by elliotwoods** [comment]
So the issue is that when you try to change, VS reports unspecified error.. Hmm. Perhaps try changing the platform toolset in the vcxproj using a text editor.

I haven't come across this issue before, it's unfortunate

Also which version of oF are you using?

**Comment by taxfromdk** [comment]
Hi Elliot,

First of all Thank you for your time. I have tried giving as much info as
possible

I am on / I did

   1. Windows 7 professional  64 bit
   2. Visual Studio Express 2012 for Windows Desktop
   3. Download openFrameworks from
   http://www.openframeworks.cc/versions/v0.8.4/of_v0.8.4_vs_release.zip
   and unzip it in a workfolder. [workfolder]\of_v0.8.4_vs_release
   4. Use project generator to create a project "mySketch" with ofxGui,
   ofxNetwork, ofxOpenCv and ofxXmlSettings
   5. Open
   [workfolder]\of_v0.8.4_vs_release\apps\myApps\mySketch\mySketch.sln and
   compile to make sure ofx can build and run.
   6. Download ofxBlackmagic2 from
   https://github.com/elliotwoods/ofxBlackmagic2/archive/master.zip and
   unzip it under [workfolder]\of_v0.8.4_vs_release\other\ofxBlackmagic2-master
   7. Open [workfolder]\of_v0.8.4_vs_release\other\ofxBlackmagic2-master\ofxBlackmagicLib\ofxBlackmagicLib.vcxproj
   using visual studio.
   8. Build

This is where it gets hairy as i get this error:


​
Now I change the Platform toolset defines insude the
ofxBlackmagicLib.vcxproj from v120 to v110 and can build further. I then
get this error.


​

I assume the last steps if this worked would be

   1. Open mySketch and add ofxBlackmagicLib.
   2. Make sure "Configuration Properties\ C/C++ \ Advanced" is set to
   Default
   3. Add ofxBlackmagiclib to the solution and set it as a build reference
   to mySketch
   4. Mimic the example usage from exampleSImple while obeying the include
   order rules.
   5. Profit!

Changing the Platform tolset to v110 got me a bit further, but im  still
not there.

I highly appreciate that you take time to support.

Kind regards

Jesper





2015-05-05 5:48 GMT+02:00 Elliot Woods <notifications@github.com>:

> So the issue is that when you try to change, VS reports unspecified
> error.. Hmm. Perhaps try changing the platform toolset in the vcxproj using
> a text editor.
>
> I haven't come across this issue before, it's unfortunate
>
> Also which version of oF are you using?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/elliotwoods/ofxBlackmagic2/issues/5#issuecomment-98934161>
> .
>



-- 
Jesper Taxbøl
+45 61627501

**Comment by taxfromdk** [comment]
I tried just commenting out the line and it could build.

Then i continued along the track and added the lib project to mySketch and
set the reference to other projects. Thev VS throws an Unspecified error,
adn when I try again aI get this.

[image: Billede indsat i tekst 1]

So I suspect the solution file is broken in a way?

If I just try to build I get

[image: Billede indsat i tekst 2]

I hope it makes sense in some way.

Kind regards

Jesper


2015-05-05 8:08 GMT+02:00 Jesper Taxbøl <jesper@taxboel.dk>:

> Hi Elliot,
>
> First of all Thank you for your time. I have tried giving as much info as
> possible
>
> I am on / I did
>
>    1. Windows 7 professional  64 bit
>    2. Visual Studio Express 2012 for Windows Desktop
>    3. Download openFrameworks from
>    http://www.openframeworks.cc/versions/v0.8.4/of_v0.8.4_vs_release.zip
>    and unzip it in a workfolder. [workfolder]\of_v0.8.4_vs_release
>    4. Use project generator to create a project "mySketch" with ofxGui,
>    ofxNetwork, ofxOpenCv and ofxXmlSettings
>    5. Open
>    [workfolder]\of_v0.8.4_vs_release\apps\myApps\mySketch\mySketch.sln and
>    compile to make sure ofx can build and run.
>    6. Download ofxBlackmagic2 from
>    https://github.com/elliotwoods/ofxBlackmagic2/archive/master.zip and
>    unzip it under [workfolder]\of_v0.8.4_vs_release\other\ofxBlackmagic2-master
>    7. Open [workfolder]\of_v0.8.4_vs_release\other\ofxBlackmagic2-master\ofxBlackmagicLib\ofxBlackmagicLib.vcxproj
>    using visual studio.
>    8. Build
>
> This is where it gets hairy as i get this error:
>
>
> ​
> Now I change the Platform toolset defines insude the
> ofxBlackmagicLib.vcxproj from v120 to v110 and can build further. I then
> get this error.
>
>
> ​
>
> I assume the last steps if this worked would be
>
>    1. Open mySketch and add ofxBlackmagicLib.
>    2. Make sure "Configuration Properties\ C/C++ \ Advanced" is set to
>    Default
>    3. Add ofxBlackmagiclib to the solution and set it as a build
>    reference to mySketch
>    4. Mimic the example usage from exampleSImple while obeying the
>    include order rules.
>    5. Profit!
>
> Changing the Platform tolset to v110 got me a bit further, but im  still
> not there.
>
> I highly appreciate that you take time to support.
>
> Kind regards
>
> Jesper
>
>
>
>
>
> 2015-05-05 5:48 GMT+02:00 Elliot Woods <notifications@github.com>:
>
>> So the issue is that when you try to change, VS reports unspecified
>> error.. Hmm. Perhaps try changing the platform toolset in the vcxproj using
>> a text editor.
>>
>> I haven't come across this issue before, it's unfortunate
>>
>> Also which version of oF are you using?
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/elliotwoods/ofxBlackmagic2/issues/5#issuecomment-98934161>
>> .
>>
>
>
>
> --
> Jesper Taxbøl
> +45 61627501
>



-- 
Jesper Taxbøl
+45 61627501

**Comment by elliotwoods** [comment]
Hi!
The images did not attach.
Perhaps you included them in email rather than attaching through the Github issues page itself?
I'll be on Windows shortly and can have a look

**Comment by taxfromdk** [comment]
Im trying again with images.

"""
Hi Elliot,

First of all Thank you for your time. I have tried giving as much info as
possible

I am on / I did

   1. Windows 7 professional  64 bit
   2. Visual Studio Express 2012 for Windows Desktop
   3. Download openFrameworks from
   http://www.openframeworks.cc/versions/v0.8.4/of_v0.8.4_vs_release.zip
   and unzip it in a workfolder. [workfolder]\of_v0.8.4_vs_release
   4. Use project generator to create a project "mySketch" with ofxGui,
   ofxNetwork, ofxOpenCv and ofxXmlSettings
   5. Open
   [workfolder]\of_v0.8.4_vs_release\apps\myApps\mySketch\mySketch.sln and
   compile to make sure ofx can build and run.
   6. Download ofxBlackmagic2 from
   https://github.com/elliotwoods/ofxBlackmagic2/archive/master.zip and
   unzip it under [workfolder]\of_v0.8.4_vs_release\other\ofxBlackmagic2-master
   7. Open [workfolder]\of_v0.8.4_vs_release\other\ofxBlackmagic2-master\ofxBlackmagicLib\ofxBlackmagicLib.vcxproj
   using visual studio.
   8. Build

This is where it gets hairy as i get this error:

![vserror](https://cloud.githubusercontent.com/assets/678546/7467927/61994412-f302-11e4-9ad6-1bb8faa17bc1.png)
​
Now I change the Platform toolset defines insude the
ofxBlackmagicLib.vcxproj from v120 to v110 and can build further. I then
get this error.

![vserror2](https://cloud.githubusercontent.com/assets/678546/7467929/6dc2902c-f302-11e4-8d8b-7da4299958d4.png)


I assume the last steps if this worked would be

   1. Open mySketch and add ofxBlackmagicLib.
   2. Make sure "Configuration Properties\ C/C++ \ Advanced" is set to
   Default
   3. Add ofxBlackmagiclib to the solution and set it as a build reference
   to mySketch
   4. Mimic the example usage from exampleSImple while obeying the include
   order rules.
   5. Profit!

Changing the Platform tolset to v110 got me a bit further, but im  still
not there.

I highly appreciate that you take time to support.

Kind regards

Jesper

"""

and

"""
I tried just commenting out the line and it could .

Then i continued along the track and added the lib project to mySketch and
set the reference to other projects. Thev VS throws an Unspecified error,
and when I try again I get this.

![vserror3](https://cloud.githubusercontent.com/assets/678546/7467936/8d5765d4-f302-11e4-80f3-e04618c001b4.png)

So I suspect the solution file is broken in a way?

If I just try to build I get

![vserror4](https://cloud.githubusercontent.com/assets/678546/7467948/b7129786-f302-11e4-9af6-583bd98f0947.png)

I hope it makes sense in some way.

Kind regards

Jesper
"""

**Comment by elliotwoods** [comment]
hey!
got the images now.
thank you for detailing your issue fully. it really helps.
having a look through now

**Comment by elliotwoods** [comment]
ok. any chance we can open a chat?
are you on Slack by any chance? or Skype?

otherwise let's continue here

the first thing you should try isn't to build the project by itself (i've never tried that, but it's anyway not really a useful thing to do). The first thing to try is to build `exampleSimple`. Can you try that and let me know your results? 

From my screen it looks like you'll just need to downgrade `ofxBlackmagicLib` to `v110` (which you've already done):
![image](https://cloud.githubusercontent.com/assets/328294/7468398/55e61720-f308-11e4-9641-ab58550a0dc9.png)

**Comment by taxfromdk** [comment]
Hi Elliot,

It works out better when I build exampleSimple, but I still get lots of include errors. I started placing it under the "Other" folder.

What path inside openframeworks should the ofxBlackmagic2 folder reside in for the includes to be correct?

**Comment by elliotwoods** [comment]
can you clarify what you mean by placing it under the "other" folder?

the path should be 
`openFrameworks/addons/ofxBlackmagic2` as with all other addons

**Comment by taxfromdk** [comment]
I was using the *"of_v0.8.4_vs_release\other"*, but that was based on
looking at broken include paths.

Do I move the exampleSimple into *"\of_v0.8.4_vs_release\apps\myApps"*

Kind regards

Jesper

2015-05-05 15:40 GMT+02:00 Elliot Woods <notifications@github.com>:

> can you clarify what you mean by placing it under the "other" folder?
>
> the path should be
> openFrameworks/addons/ofxBlackmagic2 as with all other addons
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/elliotwoods/ofxBlackmagic2/issues/5#issuecomment-99080490>
> .
>



-- 
Jesper Taxbøl
+45 61627501

**Comment by taxfromdk** [comment]
I do the following now.

Unpack openframeworks in a workfolder
Download ofxBlackmagic2-master and rename it to ofxBlackmagic and place it under addons.
Open the ofxBlackmagicLib/ofxBlackmagicLib.vcxproj and replace "v120" with "v110".
Open exampleSimple with VS 2012 Express
Press build

That leaves further than I have ever gone with this error.

![vserror](https://cloud.githubusercontent.com/assets/678546/7474199/185a9600-f340-11e4-8c3e-59fee77070b3.png)

I feel so close :)

**Comment by elliotwoods** [comment]
that error usually happens when your include order is incorrect (e.g. including ofMain before ofxBlackmagic.h)

i'm off windows for today so can't test here. should be able to have a look tomorrow

now we know you're trying the right example in the right place, we should be able to figure this out (and you shouldn't be encountering an error at this point given you're on untouched oF + ofxBlackMagic2). sorry to pause you there,

**Comment by taxfromdk** [comment]
It works!!!!!

I just commented out the function it complained about being defined twice. The problem is probably elsewhere, but I have image on screen.

Thank you so much for your help!!! :) :) :)

Jesper

**Comment by elliotwoods** [comment]
note that ofVideoPlayer might not work now you’ve commented that out though :)

**Comment by taxfromdk** [comment]
I will. 

I am following the rules about include order, so im not sure how to fix it otherwise. I hope that a fix can be found.

Jesper
